event_routing_backends/signals.py

```diff
@@ -0,0 +1,50 @@
+import json
+import logging
+
+from django.conf import settings
+from django.dispatch import receiver
+from django_redis import get_redis_connection
+from edx_django_utils.cache.utils import get_cache_key
+from openedx_events.analytics.signals import TRACKING_EVENT_EMITTED
+from openedx_events.event_bus import get_producer
+
+from event_routing_backends.config import SEND_TRACKING_EVENT_EMITTED_SIGNAL
+
+logger = logging.getLogger(__name__)
+
+TRANSFORMED_EVENT_KEY_NAME = "transformed_events"
+
+@receiver(TRACKING_EVENT_EMITTED)
+def listen_for_tracking_event_emitted_event(sender, signal, **kwargs):
+    """
+    Publish `TRACKING_EVENT_EMITTED` events to the event bus.
+    """
+    event = kwargs['tracking_log']
+
+    ## TODO: Should we ignore events that we don't care about here or in the event routing backend config?
+
+    redis = get_redis_connection("default")
+
+    queue_size = redis.llen(TRANSFORMED_EVENT_KEY_NAME)
+    if queue_size >= settings.EVENT_ROUTING_BATCH_SIZE:
+        queued_events = redis.rpop(TRANSFORMED_EVENT_KEY_NAME, settings.EVENT_ROUTING_BATCH_SIZE)
+        queued_events.append(event)
+        ## TODO: Send events to event bus in batch
+        logger.info("Sending events to event bus in batch")
+
+        #if SEND_TRACKING_EVENT_EMITTED_SIGNAL.is_enabled():
+        #    get_producer().send(
+        #        signal=TRACKING_EVENT_EMITTED,
+        #        topic='analytics',
+        #        event_key_field='tracking_log.name',
+        #        event_data={'tracking_log': kwargs['tracking_log']},
+        #        event_metadata=kwargs['metadata']
+        #    )
```

Comment:
@bmtcril I guess we would need to write the code for the event-bus consumer, but how will we send queued events in batch to the event bus?

Reply:
I don't think there's any additional code needed for the consumer. Running the `consume_events` command inside a configured edx-platform container will read events off the event bus and fire off this signal for events that match. The batching capability is in my current PR: https://github.com/openedx/event-routing-backends/pull/301/files#diff-f24cf45168ff0c054647f2bf8c504fef13b9acf37fd97eedfd7410cc61c06458R112

You would need to create an EventsRouter with the processors defined in configuration just like the AsyncRouter does (note: not like the management command does, it does not use the processors). Then you can call bulk_send on that and it will find all of the configured LRSs and POST the batch to them.